robot/Cumulus/resources/DataImportPageObject.py

```diff
@@ -9,22 +10,63 @@ class DataImportPage(ListingPage):
     def npsp(self):
         return self.builtin.get_library_instance('NPSP')
     
+    @property
+    def pageobjects(self):
+        return self.builtin.get_library_instance("cumulusci.robotframework.PageObjects")
     
-    
-        
+    def _is_current_page(self):
+        """
+        Waits for the current page to be a Data Import list view
+        """
+        self.npsp.wait_until_url_contains("/list")
+            
     def click(self,btn_name):
         """Clicks button with specified name on Data Import Page"""
         self.npsp.click_special_object_button(btn_name)
         
-    def click_begin_data_import_process(self):
-        """On the DI page, clicks the Begin Data Import Process button and waits for batch processing to complete"""
+    def begin_data_import_process_and_verify_status(self,batch,status):
+        """On the DI page, clicks the Begin Data Import Process button and waits for specified status to display """
         self.npsp.wait_for_locator("frame","NPSP Data Import")
         self.npsp.select_frame_and_click_element("NPSP Data Import","button","Begin Data Import Process")
-        self.npsp.wait_for_batch_to_complete("data_imports.status","Completed")
+        self.npsp.wait_for_batch_to_process(batch,status)
         
     def click_close_button(self):
         """Click on close button on DI processing page and waits for DI object homepage to load"""
         self.npsp.click_button_with_value("Close")
-        self.npsp.wait_until_url_contains("DataImport__c")    
+        self.npsp.wait_until_url_contains("DataImport__c")      
+        
+    def open_data_import_record(self,di_name): 
+        """Clicks on the specified data import record to open the record""" 
+        self.pageobjects.current_page_should_be("Listing","DataImport__c")
+        self.npsp.click_link_with_text(di_name)
+        self.pageobjects.current_page_should_be("Details","DataImport__c")
+        
+        
```

Comment:
There's a lot of extra whitespace in this file, as well as some other formatting issues. Could we run `black` on it?